### PR TITLE
Amiga support

### PR DESCRIPTION
**Pull request by sezero**
Amiga support

### CODE REVIEW

**Comment by sezero** [comment]
PING ?

**Comment by psi29a** [comment]
Same: https://github.com/Mindwerks/wildmidi/pull/145#issuecomment-232019848

**Comment by psi29a** [comment]
My clang caught something:
```
[ 95%] Building C object src/CMakeFiles/wildmidi.dir/wildmidi.c.o
/home/bcurtis/Workspace/Private/Wildmidi/wildmidi/src/wildmidi.c: In function ‘close_wav_output’:
/home/bcurtis/Workspace/Private/Wildmidi/wildmidi/src/wildmidi.c:300:30: warning: logical not is only applied to the left hand side of comparison [-Wlogical-not-parentheses]
 #define WM_IS_BADF(_fd) (_fd)<0
                              ^
/home/bcurtis/Workspace/Private/Wildmidi/wildmidi/src/wildmidi.c:470:10: note: in expansion of macro ‘WM_IS_BADF’
     if (!WM_IS_BADF(audio_fd))
          ^
[100%] Linking C executable ../wildmidi
```

travis's clang caught this (some thing, different warning)
https://travis-ci.org/Mindwerks/wildmidi/jobs/144175495
```
/home/bcurtis/Workspace/Private/Wildmidi/wildmidi/src/wildmidi.c:300:30: warning: logical not is only applied to the left hand side of comparison [-Wlogical-not-parentheses]
 #define WM_IS_BADF(_fd) (_fd)<0
                              ^
/home/bcurtis/Workspace/Private/Wildmidi/wildmidi/src/wildmidi.c:470:10: note: in expansion of macro ‘WM_IS_BADF’
     if (!WM_IS_BADF(audio_fd))
          ^
```

**Comment by psi29a** [comment]
Thankfully this merges in fine, but a rebase couldn't hurt either.

File: src/wildmidi_lib.c [hunk 1]
```diff
@@ -273,6 +256,48 @@ static char *wm_strdup (const char *str) {
...
+    return ((wm_isupper(c)) ? (c | ('a' - 'A')) : c);
+}
+#if 0 /* clang whines that these aren't used. */
+static inline int wm_islower(int c) {
+    return (c >= 'a' && c <= 'z');
+}
+static inline int wm_toupper(int c) {
+    return ((wm_islower(c)) ? (c & ~('a' - 'A')) : c);
+}
+#endif
```
Comment (psi29a) [inline src/wildmidi_lib.c]: If they are't used, then there is no reason to have them code right? :)

**Comment by psi29a** [comment]
Aside from clang's warnings, it looks good. I like that inlined some of functions/macros we used to import in. It makes it a heck of a lot easier to be portable. Cheers!

**Comment by sezero** [comment]
My clang didn't warn, but I pushed 62826f to hopefully fix that.

**Comment by psi29a** [comment]
Clang no longer complains.

Great job! :)

**Pull request merged**